src/main/java/gov/cabinetofice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -106,4 +122,29 @@ public User deleteUser(Integer id) {
         userRepository.delete(user);
         return user;
     }
+
+    public void invalidateUserJwt(final Cookie customJWTCookie, final HttpServletResponse response) {
+        if (customJWTCookie.getValue() != null) {
+            jwtBlacklistService.addJwtToBlacklist(customJWTCookie.getValue());
+        }
+        final Cookie userTokenCookie = WebUtil.buildCookie(
+                new Cookie(userServiceCookieName, null),
+                Boolean.TRUE,
+                Boolean.TRUE,
+                null
+        );
+        userTokenCookie.setMaxAge(0);
+        response.addCookie(userTokenCookie);
+
+        final String authenticationCookieDomain = Objects.equals(this.configProperties.getProfile(), "LOCAL") ? "localhost" : "cabinetoffice.gov.uk";
+
+        final Cookie thirdPartyAuthToken = WebUtil.buildCookie(
+                new Cookie(authenticationProvider.getTokenCookie(), null),
+                Boolean.TRUE,
+                Boolean.TRUE,
+                authenticationCookieDomain
+        );
+        thirdPartyAuthToken.setMaxAge(0);
+        response.addCookie(thirdPartyAuthToken);
```

Comment:
could use deleteCookie here as well but might need to overload it to add the domain